jcache/commons/src/main/java/org/infinispan/jcache/AbstractJCacheNotifier.java

```diff
@@ -57,6 +69,8 @@
 
    private AbstractJCacheListenerAdapter<K,V> listenerAdapter;
 
+   private ConcurrentMap<List<Object>, Queue<CountDownLatch>> latchesByEventSource = new ConcurrentHashMap<>();
```

Comment:
The key (list) contains a tuple (cache, key, value). The list is not modified after the entry is inserted in the map.

I used a list just to avoid adding a class with the sole purpose of acting as key in this map but I can add one if that's prefered approach.